ingest-app/src/cmr/ingest/data/granule_bulk_update.clj

```diff
@@ -0,0 +1,253 @@
+(ns cmr.ingest.data.granule-bulk-update
+  "Stores and retrieves granule bulk update status and task information."
+  (:require
+   [cheshire.core :as json]
+   [clojure.java.jdbc :as j]
+   [cmr.common.lifecycle :as lifecycle]
+   [cmr.common.log :refer (debug info warn error)]
+   [cmr.common.services.errors :as errors]
+   [cmr.common.util :refer [defn-timed] :as util]
+   [cmr.ingest.config :as config]
+   [cmr.oracle.connection]
+   [cmr.oracle.connection :as oracle]
+   [cmr.oracle.sql-utils :as su]))
+
+(defn- generate-failed-msg
+ "Generate the FAILED part of the status message."
+ [num-failed-granules]
+ (when (not= 0 num-failed-granules)
+   (str num-failed-granules " FAILED")))
+
+(defn- generate-skipped-msg
+ "Generate the SKIPPED part of the status message."
+ [num-failed-granules num-skipped-granules num-total-granules]
+ (when (not= 0 num-skipped-granules)
+   (if (not= 0 num-failed-granules)
+     (if (not= num-total-granules (+ num-failed-granules num-skipped-granules))
+       (str ", " num-skipped-granules " SKIPPED")
+       (str " and " num-skipped-granules " SKIPPED"))
+     (str num-skipped-granules " SKIPPED"))))
+
+(defn- generate-updated-msg
```

Comment:
should this function be in this file or in a new one. It's not related to SQL

Reply:
It's only being used in this file, so it makes sense for it to stay here, I think